R/target_market_share.R

```diff
@@ -208,24 +208,42 @@ target_market_share <- function(data,
         green_or_brown = "green_or_brown"
       )
     ) %>%
-    select(-.data$target_name, -.data$green_or_brown) %>%
+    select(-.data$target_name, -.data$green_or_brown)
+
+  data <- data %>%
+    maybe_group_by_name_ald(
+      c("sector_ald", "year", "scenario", "region"),
+      by_company = by_company
+    ) %>%
+    mutate(
+      .x = .data$weighted_production_target,
+      weighted_technology_share_target = .data$.x / sum(.data$.x),
+      .x = NULL
+    ) %>%
     pivot_wider(
       names_from = .data$scenario,
-      names_prefix = "weighted_production_target_",
-      values_from = .data$scenario_target_value,
+      values_from = c(.data$weighted_production_target, .data$weighted_technology_share_target),
       values_fn = list
     ) %>%
     tidyr::unnest(starts_with("weighted_production_")) %>%
+    tidyr::unnest(starts_with("weighted_technology_share_")) %>%
     rename(
       weighted_production_projected = .data$weighted_production,
+      weighted_technology_share_projected = .data$weighted_technology_share,
       sector = .data$sector_ald
-    ) %>%
-    pivot_longer(
-      cols = starts_with("weighted_production_"),
-      names_prefix = "weighted_production_",
-      names_to = "production_metric",
-      values_to = "production_value"
-    ) %>%
+    )
+
+  replacements <- c(
+    "weighted_" = "",
+    "(production)_" = "\\1-",
+    "(technology_share)_" = "\\1-"
+  )
+
+  data %>%
+    pivot_longer(cols = starts_with("weighted_")) %>%
+    mutate(name = stringr::str_replace_all(.data$name, replacements)) %>%
```

Comment:
To avoid stringr you can do something like this:

```r
  data %>%
    pivot_longer(cols = starts_with("weighted_")) %>%
    mutate(name = sub("weighted_", "", .data$name)) %>%
    mutate(name = sub("(production)_", "\\1-", .data$name)) %>%
    mutate(name = sub("(technology_share)_", "\\1-", .data$name)) %>%
   # ... more code

```

Reply:
Done.